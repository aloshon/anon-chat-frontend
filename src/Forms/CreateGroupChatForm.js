import React, {useContext} from "react";
import {useHistory} from "react-router-dom";
import useFields from '../Hooks/useFields';
import "./Forms.css";
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import UserContext from '../UserContext';
import {useDispatch} from 'react-redux';
import {addGroupChat} from '../actionCreators';

const CreateGroupChatForm = () => {
    const history = useHistory();
    const dispatch = useDispatch();
    const {user} = useContext(UserContext);
    const [formData, handleChange] = useFields({
        title: '',
        description: ''
    });

    const handleSubmit = async (e) => {
        e.preventDefault();
        try{
            formData.creator_id = user.id;
            formData.creatorToGuestList = {id: user.id, username: user.username};
            // allow the dipatched function to finish
            await dispatch(addGroupChat(formData));
            history.push('/');
        } catch(e){
            console.log(`Error creating group chat!:${e}`)
            alert(`Error creating group chat!:${e}`);
        }
    }

    return (
        <Container className="createform">
            <h2 className="create-chat-header">Create New anonChat</h2>
            <Form onSubmit={handleSubmit}>
                <Form.Group>
                    <Form.Label>Title:</Form.Label>
                    <Form.Control name="title" type="text" maxLength="100" value={formData.title} onChange={handleChange} placeholder="Title" />
                </Form.Group>
                <Form.Group>
                    <Form.Label>Description:</Form.Label>
                    <Form.Control name="description" type="text" maxLength="400" value={formData.description} onChange={handleChange} placeholder="Description" />
                </Form.Group>
                <Button variant="primary" type="submit">
                    Submit
                </Button>
                <Button variant="secondary" onClick={() => history.push('/')} >
                    Back
                </Button>
            </Form>
        </Container>
    )
}

export default CreateGroupChatForm;