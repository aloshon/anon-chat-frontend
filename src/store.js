import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);