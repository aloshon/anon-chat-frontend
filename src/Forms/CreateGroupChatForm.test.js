import { render, screen } from '@testing-library/react';
import CreateGroupChatForm from './CreateGroupChatForm';
import { Provider } from "react-redux";
import { store } from "../store"
import {BrowserRouter} from 'react-router-dom';
import UserContext from "../UserContext";


describe("CreateGroupChatForm should render form for user to create group chats", () => {
    const user = {
        id: 1,
        username: "user1",
        isAdmin: false,
        blockList: []
    }

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <UserContext.Provider value={{user}}>
                <CreateGroupChatForm />
            </UserContext.Provider>
        </BrowserRouter>
      </Provider>
    );
    test("renders labels", () => {
      const label1 = screen.getByText("Title:");
      const label2 = screen.getByText("Description:");
      const button = screen.getByText("Submit");
      expect(label1).toBeInTheDocument();
      expect(label2).toBeInTheDocument();
      expect(button).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });