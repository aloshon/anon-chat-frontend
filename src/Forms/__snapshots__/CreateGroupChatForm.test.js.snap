// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroupChatForm should render form for user to create group chats matches snapshot 1`] = `<DocumentFragment />`;
